src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs

```diff
@@ -635,8 +690,11 @@ internal GraphServiceClient GetGraphServiceClient(TokenCredential tokenCredentia
             effectiveApplicationParameters.Instance = isB2C
                 ? $"https://{effectiveApplicationParameters.Domain1}.b2clogin.com/"
                 : originalApplicationParameters.Instance;
-
-            effectiveApplicationParameters.PasswordCredentials.AddRange(application.PasswordCredentials.Select(p => p.Hint + "******************"));
+            if (effectiveApplicationParameters.PasswordCredentials is List<string>)
```

Comment:
pattern matching (... is List<string> passwordCredentials), and not duplicate the cast below?  Also, should it be IList, in case it's not that particular concrete implementation in the future?

Reply:
the cast isn't needed at all for now. Not really needed for it to be an IList since we instantiate and populate the List ourselves manually. Removed the cast.